src/data.py

```diff
@@ -59,13 +59,9 @@ def get_image_paths(self, dataset_dir, split):
 
 class EncodedVisualNewsDataset(Dataset):
 
-    def __init__(self,
-                 visual_news_dataset_dir,
-                 encoded_data_file,
-                 split,
-                 headlines=False):
-        self.data = self.get_data(visual_news_dataset_dir, encoded_data_file,
-                                  split, headlines)
+    def __init__(self, encoded_dataset, split):
```

Comment:
Yes, this way we are able to remove the logic out of dataset class abut whether to use captions or headlines. It's better to handle that via command line arguments which simplifies the dataset class.

Also, since each encoded file is supposed to only contain captions or headlines, it reduces the time and memory it takes to load the entire data file in memory.